packages/connection/src/class-client.php

```diff
@@ -369,6 +369,13 @@ public static function wpcom_json_api_request_as_blog(
 		$body = null,
 		$base_api_path = 'rest'
 	) {
+
+		// Allow use a store sandbox. Internal ref: PCYsg-IA-p2.
+		if ( isset( $_COOKIE ) && isset( $_COOKIE['store_sandbox'] ) ) {
+			$secret                    = $_COOKIE['store_sandbox'];
+			$args['headers']['Cookie'] = "store_sandbox=$secret;";
```

Comment:
This is basically copy-paste solution from another file. This approach will override Cookie value, which is not good, right?

Reply:
Good catch.
I modified the code to prevent the cookies from being overwritten.
We probably need to switch to using `WP_Http_Cookie` at some point, but I doubt it will happen anytime soon because these two approaches aren't compatible.

Reply:
Thanks!